### PR TITLE
Generates SBOM after source files are updated with release tag

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
Updates the auto release script to do the release tag updation in source files before SBOM generation.

Test Steps
-----------
Tested in fork

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aggarg — APPROVED** [review]

File: .github/workflows/auto-release.yml [hunk 1]
```diff
@@ -25,7 +25,6 @@ jobs:
       - name: Tool Setup
         uses: actions/setup-python@v2
         with:
-          python-version: 3.7.10
```
Comment (amazonKamath) [inline .github/workflows/auto-release.yml:28]: why did we remove the python version

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -51,7 +50,18 @@ jobs:
...
+      - name: Update source files with version info
+        run: |
+          # Install deps and run
+          pip install -r ./tools/.github/scripts/release-requirements.txt
+          ./tools/.github/scripts/update_src_version.py FreeRTOS --kernel-repo-path=local_kernel --kernel-commit=${{ env.COMMIT_SHA_1 }} --new-kernel-version=${{ github.event.inputs.version_number }} --new-kernel-main-br-version=${{ github.event.inputs.main_br_version }}
+          exit $?
+        env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Generate SBOM
```
Comment (amazonKamath) [inline .github/workflows/auto-release.yml:66]: not really generating SBOM is it ?

**Review by amazonKamath — APPROVED** [review]

File: .github/workflows/auto-release.yml (hunk 1 above)
Comment (tony-josi-aws) [reply .github/workflows/auto-release.yml:28]: That version of python was no longer available in the system, also I didn't find any dependency on the python version while testing.

File: .github/workflows/auto-release.yml (hunk 2 above)
Comment (tony-josi-aws) [reply .github/workflows/auto-release.yml:66]: It creates a SBOM file and keeps it in the source folder, but I'm not sure if generate is the right word.

**Pull request merged**